plugins/field-colour/src/blocks/colourBlend.ts

```diff
@@ -0,0 +1,271 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly';
+import * as JavaScript from 'blockly/javascript';
+import * as Dart from 'blockly/dart';
+import * as Lua from 'blockly/lua';
+import * as PHP from 'blockly/php';
+import * as Python from 'blockly/python';
+import { registerColourField } from '../field_colour';
+import { Generators, installGenerators } from './generatorUtils';
+
+
+const blockName = 'colour_blend';
+
+// Block for blending two colours together.
+const jsonDef =
+{
+    'type': blockName,
+    'message0':
+        '%{BKY_COLOUR_BLEND_TITLE} %{BKY_COLOUR_BLEND_COLOUR1} ' +
+        '%1 %{BKY_COLOUR_BLEND_COLOUR2} %2 %{BKY_COLOUR_BLEND_RATIO} %3',
+    'args0': [
+        {
+            'type': 'input_value',
+            'name': 'COLOUR1',
+            'check': 'Colour',
+            'align': 'RIGHT',
+        },
+        {
+            'type': 'input_value',
+            'name': 'COLOUR2',
+            'check': 'Colour',
+            'align': 'RIGHT',
+        },
+        {
+            'type': 'input_value',
+            'name': 'RATIO',
+            'check': 'Number',
+            'align': 'RIGHT',
+        },
+    ],
+    'output': 'Colour',
+    'helpUrl': '%{BKY_COLOUR_BLEND_HELPURL}',
+    'style': 'colour_blocks',
+    'tooltip': '%{BKY_COLOUR_BLEND_TOOLTIP}',
+};
+
+
+/**
+ * Javascript generator definition.
+ * @param block
+ * @param generator 
+ * @returns 
+ */
+export function jsGenerator(
```

Comment:
I am okay with `blockToJavascript`. 

I would be extremely unhappy with `javascript`, because even in context I don't think a variable named `javascript` or `dart` or `python` clearly indicates what it is and how it is used.

I'm neutral on `javascriptBlockGenerator`.